### PR TITLE
feat(workflow): Unified alert rules list

### PR DESCRIPTION
**Pull request by scttcper**
- Adds an org-wide alert rules list page that unifies metric and issue alert rules. Consumes from https://github.com/getsentry/sentry/pull/19832
- Removes the alert rules list at the individual project settings level
- Changes the landing page when navigating from the side nav. If a user has metric alerts it is still the alert stream, if they do not it is the rules list page. Side effect - removes the current metric alert upsell page, will be repositioned in a follow up.

image

### CODE REVIEW

File: src/sentry/static/sentry/app/views/alerts/list/index.tsx [hunk 1]
```diff
@@ -277,70 +255,28 @@ class IncidentsList extends AsyncComponent<Props, State & AsyncComponent['state'
     const status = getQueryStatus(query.status);
 
     return (
-      <DocumentTitle title={`Alerts- ${orgId} - Sentry`}>
+      <DocumentTitle title={`Alerts - ${orgId} - Sentry`}>
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/alerts/list/index.tsx:258]: `<SentryDocumentTitle>` might be preferred here so you don't have to worry about the formatting and can add i18n
```suggestion
      <SentryDocumentTitle title={t('Alerts')} objSlug={organization.slug}>
```

File: src/sentry/static/sentry/app/views/alerts/rules/index.tsx [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    );
+  }
+
+  handleDeleteRule = async (projectId: string, ruleId: string) => {
+    const {params} = this.props;
+    const {orgId} = params;
+
+    try {
+      await this.api.requestPromise(
+        `/projects/${orgId}/${projectId}/alert-rules/${ruleId}/`,
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/alerts/rules/index.tsx:85]: Was just testing this out and you may need to provide a different endpoint depending on deleting a metric alert vs issue alert.

Deleting an issue alert would be at
`DELETE: /projects/${orgId}/${projectId}/rules/${ruleId}/`

Comment (scttcper) [reply src/sentry/static/sentry/app/views/alerts/rules/index.tsx]: nice catch! thanks for testing

File: src/sentry/static/sentry/app/views/alerts/list/index.tsx [hunk 3]
```diff
@@ -277,74 +255,32 @@ class IncidentsList extends AsyncComponent<Props, State & AsyncComponent['state'
     const status = getQueryStatus(query.status);
 
     return (
-      <DocumentTitle title={`Alerts- ${orgId} - Sentry`}>
+      <SentryDocumentTitle title="Alerts" objSlug={orgId}>
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/alerts/list/index.tsx:258]: nit: add i18n

```suggestion
      <SentryDocumentTitle title={t('Alerts')} objSlug={orgId}>
```

**Review by davidenwang — APPROVED** [review]
lgtm

File: src/sentry/static/sentry/app/views/alerts/rules/index.tsx [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+              <div>{t('Type')}</div>
+              <div>{t('Alert Name')}</div>
+              <div>{t('Project')}</div>
+              <div>{t('Created By')}</div>
+              <div>
+                <StyledSortLink
+                  to={{
+                    pathname: `/organizations/${orgId}/alerts/rules/`,
+                    query: {
+                      asc: sort.asc ? undefined : '1',
```
Comment (iProgramStuff) [inline src/sentry/static/sentry/app/views/alerts/rules/index.tsx:130]: I think here you will need to keep the original parameters in the query.

Tested this locally with a project filter active, the page load sent a backend query with project=1, but when I sorted it just sent asc=1 and I got rules for all projects.

**Review by iProgramStuff — DISMISSED** [review]
Looks generally good to me other than the sorting issue causing us to lose other parameters (like project filters)

File: src/sentry/static/sentry/app/views/settings/projectAlerts/settings.jsx [hunk 5]
```diff
@@ -77,30 +77,18 @@ class ProjectAlertSettings extends AsyncView {
...
-                  ? t('You do not have permission to edit alert rules.')
-                  : undefined
-              }
-              priority="primary"
-              size="small"
-              icon={<IconAdd size="xs" isCircled />}
-            >
-              {t('New Alert Rule')}
+            <Button to={`/organizations/${orgId}/alerts/rules/`} size="small">
+              {t('View Alert Rules')}
```
Comment (iProgramStuff) [inline src/sentry/static/sentry/app/views/settings/projectAlerts/settings.jsx:86]: I was wondering about whether it'd be desired to retain the project filter here. Did you talk to Adhi or anyone else about this?

Comment (scttcper) [reply src/sentry/static/sentry/app/views/settings/projectAlerts/settings.jsx:86]: we generally keep the project filter when moving between say the issues tab and discover, I think we should expect it to do the same here.

File: src/sentry/static/sentry/app/views/alerts/rules/index.tsx (hunk 4 above)
Comment (scttcper) [reply src/sentry/static/sentry/app/views/alerts/rules/index.tsx:131]: thanks, good catch

**Review by iProgramStuff — APPROVED** [review]

**Pull request merged**